From: openKylin Developers <packaging@lists.openkylin.top>
Date: Mon, 1 Jun 2026 14:28:32 +0800
Subject: Allow systemd interaction

Origin: upstream, https://gitlab.com/apparmor/apparmor/-/merge_requests/236,
 https://gitlab.com/apparmor/apparmor/-/merge_requests/811
Bug: https://gitlab.com/apparmor/apparmor/-/issues/203
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/apparmor/+bug/1952242
Applied-Upstream: https://gitlab.com/apparmor/apparmor/-/commit/924d4e87adbf02cafa169b3eed7892d0d35a2088,
 https://gitlab.com/apparmor/apparmor/-/commit/ee9e61aad284f4edbebbd7cd0e8d9ac452455958
Last-Update: 2021-11-26

When started by systemd, we need to allow notify access, and access to
certain /proc areas.
Last-Update: 2021-11-26
---
 profiles/apparmor.d/usr.sbin.smbd | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/profiles/apparmor.d/usr.sbin.smbd b/profiles/apparmor.d/usr.sbin.smbd
index b54b4f5..b105a91 100644
--- a/profiles/apparmor.d/usr.sbin.smbd
+++ b/profiles/apparmor.d/usr.sbin.smbd
@@ -27,12 +27,22 @@ profile smbd /usr/{bin,sbin}/smbd {
 
   signal send set=term peer=samba-bgqd,
 
+  # when started by systemd
+  ptrace read peer=unconfined,
+
   /etc/mtab r,
   /etc/netgroup r,
   /etc/printcap r,
   /etc/samba/* rwk,
   @{PROC}/@{pid}/mounts r,
   @{PROC}/sys/kernel/core_pattern r,
+
+  # https://gitlab.com/apparmor/apparmor/-/issues/203
+  # needed when smbd is started by systemd
+  @{PROC}/1/environ r,
+  @{PROC}/cmdline r,
+  @{PROC}/sys/kernel/osrelease r,
+
   /usr/lib*/samba/vfs/*.so mr,
   /usr/lib*/samba/auth/*.so mr,
   /usr/lib*/samba/charset/*.so mr,
@@ -57,6 +67,8 @@ profile smbd /usr/{bin,sbin}/smbd {
   @{run}/samba/** rk,
   @{run}/samba/ncalrpc/ rw,
   @{run}/samba/ncalrpc/** rw,
+  # when started by systemd
+  @{run}/systemd/notify w,
   /var/spool/samba/** rw,
 
   @{HOMEDIRS}/** lrwk,
